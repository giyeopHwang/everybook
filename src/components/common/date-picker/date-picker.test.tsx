import { fireEvent, render, screen } from '@testing-library/react';
import DatePicker from './date-picker.component';

describe('DatePicker', () => {
  it('renders DatePicker component', () => {
    const datePicker = render(<DatePicker onChange={() => {}} />);
    expect(datePicker).toMatchSnapshot();

    const resetButton = screen.getByText('초기화');
    expect(resetButton).toBeInTheDocument();
  });

  it('renders with specified date', () => {
    const datePicker = render(
      <DatePicker date={new Date('2024-01-01')} onChange={() => {}} />
    );
    expect(datePicker).toMatchSnapshot();

    const yearValue = screen.getByDisplayValue('2024');
    expect(yearValue).toBeInTheDocument();

    const monthValue = screen.getByDisplayValue('1');
    expect(monthValue).toBeInTheDocument();
  });

  it('updates and resets input values', () => {
    render(<DatePicker date={new Date('2024-01-01')} onChange={() => {}} />);

    const yearInput = screen.getByTestId('date-picker-year');
    fireEvent.change(yearInput, { target: { value: '2023' } });
    fireEvent.blur(yearInput);

    const monthInput = screen.getByTestId('date-picker-month');
    fireEvent.change(monthInput, { target: { value: '12' } });
    fireEvent.blur(monthInput);

    let yearValue = screen.getByDisplayValue('2023');
    expect(yearValue).toBeInTheDocument();

    let monthValue = screen.getByDisplayValue('12');
    expect(monthValue).toBeInTheDocument();

    const resetButton = screen.getByText('초기화');
    fireEvent.click(resetButton);

    yearValue = screen.getByDisplayValue('2024');
    expect(yearValue).toBeInTheDocument();

    monthValue = screen.getByDisplayValue('1');
    expect(monthValue).toBeInTheDocument();
  });

  it('fires callback when blurred with valid values', () => {
    const mock = vi.fn();
    render(<DatePicker onChange={mock} />);

    const yearInput = screen.getByTestId('date-picker-year');
    fireEvent.change(yearInput, { target: { value: '2023' } });
    fireEvent.blur(yearInput);

    const monthInput = screen.getByTestId('date-picker-month');
    fireEvent.change(monthInput, { target: { value: '12' } });
    fireEvent.blur(monthInput);

    expect(mock).toHaveBeenCalledWith(2023, 12);
  });

  it('does not fire callback when blurred with invalid values', () => {
    const mock = vi.fn();
    render(<DatePicker onChange={mock} />);

    const yearInput = screen.getByTestId('date-picker-year');
    fireEvent.change(yearInput, { target: { value: '3000' } });
    fireEvent.blur(yearInput);

    const monthInput = screen.getByTestId('date-picker-month');
    fireEvent.change(monthInput, { target: { value: '13' } });
    fireEvent.blur(monthInput);

    expect(mock).not.toBeCalled();
  });
});
