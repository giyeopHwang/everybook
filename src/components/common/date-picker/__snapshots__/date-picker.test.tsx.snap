// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatePicker > renders DatePicker component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1gzecui"
      >
        <div
          class="css-1gzecui"
        >
          <input
            class="css-166c21b"
            data-testid="date-picker-year"
            max="2100"
            min="1900"
            name="year"
            step="1"
            type="input"
            value="2024"
          />
          <label
            for="year"
          >
            년
          </label>
          <input
            class="css-166c21b"
            data-testid="date-picker-month"
            max="12"
            min="1"
            name="month"
            step="1"
            type="input"
            value="1"
          />
          <label
            for="month"
          >
            월
          </label>
          <button
            class="css-wieti"
          >
            초기화
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1gzecui"
    >
      <div
        class="css-1gzecui"
      >
        <input
          class="css-166c21b"
          data-testid="date-picker-year"
          max="2100"
          min="1900"
          name="year"
          step="1"
          type="input"
          value="2024"
        />
        <label
          for="year"
        >
          년
        </label>
        <input
          class="css-166c21b"
          data-testid="date-picker-month"
          max="12"
          min="1"
          name="month"
          step="1"
          type="input"
          value="1"
        />
        <label
          for="month"
        >
          월
        </label>
        <button
          class="css-wieti"
        >
          초기화
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DatePicker > renders with specified date 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1gzecui"
      >
        <div
          class="css-1gzecui"
        >
          <input
            class="css-166c21b"
            data-testid="date-picker-year"
            max="2100"
            min="1900"
            name="year"
            step="1"
            type="input"
            value="2024"
          />
          <label
            for="year"
          >
            년
          </label>
          <input
            class="css-166c21b"
            data-testid="date-picker-month"
            max="12"
            min="1"
            name="month"
            step="1"
            type="input"
            value="1"
          />
          <label
            for="month"
          >
            월
          </label>
          <button
            class="css-wieti"
          >
            초기화
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1gzecui"
    >
      <div
        class="css-1gzecui"
      >
        <input
          class="css-166c21b"
          data-testid="date-picker-year"
          max="2100"
          min="1900"
          name="year"
          step="1"
          type="input"
          value="2024"
        />
        <label
          for="year"
        >
          년
        </label>
        <input
          class="css-166c21b"
          data-testid="date-picker-month"
          max="12"
          min="1"
          name="month"
          step="1"
          type="input"
          value="1"
        />
        <label
          for="month"
        >
          월
        </label>
        <button
          class="css-wieti"
        >
          초기화
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
